src/main/java/org/threadly/concurrent/AbstractPriorityScheduler.java

```diff
@@ -507,6 +507,16 @@ public void cancel() {
     public abstract boolean canExecute(short executeReference);
     
     /**
+     * Simple getter for the run time, this is expected to do NO operations for calculating the 
+     * run time.  The main reason this is used over {@link #getRunTime()} is to allow the JVM to 
+     * jit the function better.  Because of the nature of this, this can only be used at very 
+     * specific points in the tasks lifecycle, and can not be used for sorting operations.
+     * 
+     * @return An un-molested representation of the stored absolute run time
+     */
+    protected abstract long getPureRunTime();
```

Comment:
This was a small bit of added code to keep the scheduled and recurring tasks just as performant.